runstatus.go

```diff
@@ -0,0 +1,357 @@
+type RunstatusService struct {
+	Name  string `json:"name"`
+	State string `json:"state,omitempty"`
+	URL   string `json:"url,omitempty"`
+}
+
+// RunstatusServiceList service list
+type RunstatusServiceList struct {
+	Services []RunstatusService `json:"services"`
+}
+
+// ID give the service ID
+func (service *RunstatusService) ID() (int, error) {
+	url := strings.TrimRight(service.URL, "/")
+	urlSplited := strings.Split(url, "/")
+	id, err := strconv.ParseInt(urlSplited[len(urlSplited)-1], 10, 32)
+	if err != nil {
+		return 0, err
+	}
+
+	return int(id), nil
+}
+
+// DeleteRunstatusService delete runstatus service
+func (client *Client) DeleteRunstatusService(ctx context.Context, page string, id int) error {
+	_, err := client.runstatusRequest(ctx, fmt.Sprintf("/pages/%s/services/%d", page, id), nil, "DELETE")
+	return err
+}
+
+// CreateRunstatusService create runstatus service
+func (client *Client) CreateRunstatusService(ctx context.Context, page string, service RunstatusService) error {
+	_, err := client.runstatusRequest(ctx, "/pages/"+page+"/services", service, "POST")
+	return err
+}
+
+// ListRunstatusService list runstatus service
+func (client *Client) ListRunstatusService(ctx context.Context, page string) ([]RunstatusService, error) {
+	resp, err := client.runstatusRequest(ctx, "/pages/"+page+"/services", nil, "GET")
+	if err != nil {
+		return nil, err
+	}
+
+	var p *RunstatusServiceList
+	if err := json.Unmarshal(resp, &p); err != nil {
+		return nil, err
+	}
+
+	return p.Services, nil
+}
+
+// CreateRunstatusEvent create runstatus incident event
+func (client *Client) CreateRunstatusEvent(ctx context.Context, page string, incidentID int, event RunstatusEvent) error {
+	_, err := client.runstatusRequest(ctx, fmt.Sprintf("/pages/%s/incidents/%d/events", page, incidentID), event, "POST")
+	return err
+}
+
+// ListRunstatusMaintenance list runstatus Maintenance
+func (client *Client) ListRunstatusMaintenance(ctx context.Context, page string) ([]RunstatusMaintenance, error) {
+	resp, err := client.runstatusRequest(ctx, "/pages/"+page+"/maintenances", nil, "GET")
+	if err != nil {
+		return nil, err
+	}
+
+	var p *RunstatusMaintenanceList
+	if err := json.Unmarshal(resp, &p); err != nil {
+		return nil, err
+	}
+
+	return p.Maintenances, nil
+}
+
+// CreateRunstatusMaintenance create runstatus Maintenance
+func (client *Client) CreateRunstatusMaintenance(ctx context.Context, page string, maintenance RunstatusMaintenance) error {
+	_, err := client.runstatusRequest(ctx, "/pages/"+page+"/maintenances", maintenance, "POST")
+	return err
+}
+
+// DeleteRunstatusMaintenance delete runstatus Maintenance
+func (client *Client) DeleteRunstatusMaintenance(ctx context.Context, page string, id int) error {
+	_, err := client.runstatusRequest(ctx, fmt.Sprintf("/pages/%s/maintenances/%d", page, id), nil, "DELETE")
+	return err
+}
+
+// ID give the maintenance ID
+func (service *RunstatusMaintenance) ID() (int, error) {
```

Comment:
I'd name the receiver variable `m` or `maintenance` instead of `service`, since its type is `RunstatusMaintenance`.

Reply:
👍